File: Practice/14/Python/14.py
def recf(n):
    counter = 1
    factor = 2
   
    while factor<=n:
        factor *= 2
        counter +=1
    return counter;
   

n = int(input("Введите число\n"))
if n==0: print("Количество степеней двойки: 0")
elif n<0: print ("Неверные данные.")
else: print( "Количество степеней двойки: ", recf(n))

File: Practice/14/C++/14/14.cpp
#include <iostream>
using namespace std;

int recf(int n)
{
	int counter = 1;
	int factor = 2;

	if (n == 0) return 0;

	while (factor <= n) {
		factor *= 2;
		counter +=1;
	}
	return counter;
}

int main()
{
	setlocale(LC_ALL, "russian");
	int n;

	cout << "Введите число\n";
	cin >> n;
	if (n < 0) cout << "Неверные данные.";
	else cout << "Количество степеней двойки: "<< recf(n) << '\n';

	return 0;
}